client/post-editor/editor-drawer/taxonomies.jsx

```diff
@@ -24,35 +25,26 @@ function EditorDrawerTaxonomies( { siteId, postType, postTerms, taxonomies } ) {
 				<QueryTaxonomies { ...{ siteId, postType } } />
 			) }
 			{ map( taxonomies, ( taxonomy ) => {
-				if ( 'post_format' === taxonomy.name ) {
+				const { name, label, hierarchical } = taxonomy;
+				if ( 'post_format' === name ) {
 					// Post format has its own dedicated accordion
 					return;
 				}
 
-				if ( taxonomy.hierarchical ) {
-					return (
-						<Accordion
-							key={ taxonomy.name }
-							title={ taxonomy.label }
-							icon={ <Gridicon icon="category" /> }
-						>
-							Work in Progress
-						</Accordion>
-					);
-				} else {
-					return (
-						<Accordion
-							key={ taxonomy.name }
-							title={ taxonomy.label }
-							icon={ <Gridicon icon="tag" /> }
-						>
-							<TermTokenField
-								postTerms={ postTerms }
-								taxonomyName={ taxonomy.name }
-							/>
-						</Accordion>
-					);
-				}
+				const icon = hierarchical ? 'folder' : 'tag';
+
+				return (
+					<Accordion
+						key={ name }
+						title={ label }
+						icon={ <Gridicon icon={ icon } /> }
+					>
+					{ hierarchical
```

Comment:
This would interfere with #6548 which removes the `postTerms` prop from `TermTokenField`.